scalding-spark/src/main/scala/com/twitter/scalding/spark_backend/SparkBackend.scala

```diff
@@ -11,6 +11,38 @@ import scala.collection.mutable.{ Map => MMap, ArrayBuffer }
 
 object SparkPlanner {
   import SparkMode.SparkConfigMethods
+  sealed trait PartitionComputer {
+    def apply(currentNumPartitions: Int): Int
+  }
+
+  final case object IdentityPartitionComputer extends PartitionComputer {
+    def apply(currentNumPartitions: Int): Int = currentNumPartitions
+  }
+  final case class ConfigPartitionComputer(config: Config, scaldingReducers: Option[Int]) extends PartitionComputer {
+    def apply(currentNumPartitions: Int): Int = {
+      val maxPartitions = config.getMaxPartitionCount
+      val getReducerScaling = config.getReducerScaling.getOrElse(1.0D)
+      val candidates = scaldingReducers match {
+        case None =>
+          currentNumPartitions
+        case Some(i) if i < 0 =>
+          currentNumPartitions
+        case Some(red) =>
+          (getReducerScaling * red).toInt
+      }
+      if (candidates > 0) {
+        maxPartitions match {
+          case Some(maxP) =>
+            Math.min(maxP, candidates)
+          case None =>
+            candidates
+        }
+      } else {
+        1
```

Comment:
The fall through should probably be to leave it unchanged?

Reply:
this is a case of `candidates <= 0` which can arise from bad configs from getReducerScaling.

Maybe `candidates == 0` goes to 1, else error? Negative shouldn't happen.